extensions/vertx-web-common/deployment/src/main/java/io/quarkus/vertx/web/common/deployment/VertxWebCommonProcessor.java

```diff
@@ -0,0 +1,83 @@
+package io.quarkus.vertx.web.common.deployment;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+import io.quarkus.kubernetes.spi.KubernetesPortBuildItem;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.vertx.common.deployment.InternalVertxBuildItem;
+import io.quarkus.vertx.web.common.runtime.HttpConfiguration;
+import io.quarkus.vertx.web.common.runtime.RouterProducer;
+import io.quarkus.vertx.web.common.runtime.VertxWebCommonRecorder;
+import io.quarkus.vertx.web.common.runtime.cors.CORSRecorder;
+import io.vertx.ext.web.Router;
+
+class VertxWebCommonProcessor {
+
+    HttpConfiguration httpConfiguration;
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    FilterBuildItem cors(CORSRecorder recorder, HttpConfiguration configuration) {
+        return new FilterBuildItem(recorder.corsHandler(configuration));
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem additionalBeans() {
+        return AdditionalBeanBuildItem.unremovableOf(RouterProducer.class);
+    }
+
+    @BuildStep(onlyIf = IsNormal.class)
+    @Record(value = ExecutionTime.RUNTIME_INIT, optional = true)
+    public KubernetesPortBuildItem kubernetes(HttpConfiguration config, BuildProducer<KubernetesPortBuildItem> portProducer,
+            VertxWebCommonRecorder recorder) {
+        int port = ConfigProvider.getConfig().getOptionalValue("quarkus.http.port", Integer.class).orElse(8080);
+        recorder.warnIfPortChanged(config, port);
+        return new KubernetesPortBuildItem(config.port, "http");
```

Comment:
More a question than a comment, should the https configuration be generated?

Reply:
Please ask the question to @geoand . It shouldn't be part of this PR.

Reply:
Unless I am missing something, I don't think so - this is only needed when the `kubernetes` extension is active in order to be able to populate the kubernetes resources with the proper port (the one that the application is configured to use)